eng/pipelines/templates/steps/publish-cli-choco.yml

```diff
@@ -0,0 +1,62 @@
+parameters:
+  - name: AllowPrerelease
+    type: boolean
+    default: false
+  - name: CliVersion
+    type: string
+    default: $(CLI_VERSION)
+  - name: ChocoApiKey
+    type: string
+    default: $(azure-sdk-choco-api-key)
+
+steps:
+  - task: DownloadPipelineArtifact@2
+    inputs:
+      artifactName: signed-win
+      itemPattern: azd-windows-amd64.msi
+      targetPath: cli/installer/choco/tools
+
+  - task: PowerShell@2
+    displayName: Set Choco Package Version
+    inputs:
+      pwsh: true
+      targetType: filePath
+      filePath: eng/scripts/Get-MsiVersion.ps1
+      arguments: >-
+        -Version '${{ parameters.CliVersion }}'
+        -DevOpsOutput
+
+  - pwsh: |
+      $submitPackage = eng/scripts/Test-ShouldReleasePackageVersion.ps1 `
+        -Version '$(MSI_VERSION)' `
+        -AllowPrerelease:$${{ parameters.AllowPrerelease }}
+
+      if ('$(Skip.ReleaseWinGet)' -eq 'true') {
+        $submitPackage = 'false'
+      }
+      Write-Host "Setting SubmitChocoPackage to $submitPackage"
+      Write-Host "##vso[task.setvariable variable=SubmitChocoPackage]$submitPackage"
+    displayName: Set SubmitChocoPackage
+
+  - task: PowerShell@2
+    displayName: Build Choco Package
+    inputs:
+      pwsh: true
+      targetType: filePath
+      filePath: eng/scripts/New-ChocoPackage.ps1
+      arguments: >-
+        -Version $(MSI_VERSION)
+        -Tag "azure-dev-cli_${{ parameters.CliVersion }}"
+
```

Comment:
Nit: quoting is all over the place. Above you use single quotes for a parameter where there's no quotes here, and double quotes on -Tag where you use single quotes.

In general, I recommend single quotes in PowerShell always when you don't want to expand variables. Of course, here, those variables are expended not by PowerShell (or shouldn't be), but by the pipeline template engine. So single quotes are better. Write-Host on line 56 could also be single quotes, but you take no variables so it doesn't really matter. Mostly, just use single quotes where you might take a build-time variable and definitely dont want PowerShell itself to expand variables. Also good to quote parameters that might - even erroneously - be null/empty (same thing in PowerShell like `$(MSI_VERSION)` such that an empty - probably invalid - value (you should check / author it as mandatory, which an empty string won't satisfy).